packages/web-react/src/hooks/useDeprecatedMessage.ts

```diff
@@ -25,3 +29,62 @@ export const useDeprecatedMessage = ({
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, []);
 };
+
+export interface UseDeprecationMessageProps {
+  method?: 'component' | 'property';
+  trigger: boolean;
+  componentName: string;
+  componentProps?: {
+    newName?: string;
+    delete?: boolean;
+  };
+  propertyProps?: {
+    deprecatedName?: string;
+    newName?: string;
+    delete?: boolean;
+  };
+}
+
+export const useDeprecationMessage = ({
+  method = 'component',
+  trigger,
+  componentName,
+  componentProps,
+  propertyProps,
+}: UseDeprecationMessageProps) => {
+  let message: string;
+  let condition: boolean;
+
+  useEffect(() => {
+    const run = trigger && componentName && process.env.NODE_ENV === 'development';
+
+    switch (method) {
+      case 'property':
+        if (propertyProps?.delete) {
+          message = `Deprecation warning (${componentName}): "${propertyProps?.deprecatedName}" property will be deleted in next major version..️️`;
+        } else {
+          message = `Deprecation warning (${componentName}): "${propertyProps?.deprecatedName}" property will be replaced in next major version. Please use "${propertyProps?.newName}" instead. ♻️️`;
+        }
+        condition = !!propertyProps;
+        break;
+
+      case 'component':
+      default:
+        if (componentProps?.delete) {
+          message = `Deprecation warning (${componentName}): The component and its sub components will be deleted in next major version.`;
+        } else {
+          message = `Deprecation warning (${componentName}): The component and its sub components will be renamed to "${componentProps?.newName}" in next major version.`;
+        }
+        condition = !!componentProps;
+        break;
+    }
+
+    if (message && run && condition) {
```

Comment:
Please name properly the `run` and `condition` variables. I suggest that `run` should be something like `isDispatchable` or `isFirable`. And `condition` is too general. Also why is the condition optional if you, based on the content of the variable, fire or not the console.log?

If I understand this correctly then if I do not fill componentProps or propertyProps, the message will be never fired. So maybe it is better to make these props required.